kubetest/e2e.go

```diff
@@ -615,33 +615,53 @@ func kubemarkUp(dump string, o options, deploy deployer) error {
 		return err
 	}
 
-	masterIP, err := control.Output(exec.Command(
-		"gcloud", "compute", "addresses", "describe",
-		os.Getenv("MASTER_NAME")+"-ip",
-		"--project="+o.gcpProject,
-		"--region="+o.gcpZone[:len(o.gcpZone)-2],
-		"--format=value(address)"))
-	if err != nil {
-		return fmt.Errorf("failed to get masterIP: %v", err)
+	var masterIP, masterInternalIP []byte
+
+	switch o.deployment {
+	case "gce":
```

Comment:
existing kubemark tests don't set --deployment=gce (it's not even correct value of deployment flag). We use --deployment=bash with --provider=gce

This breaks our kubermark tests in gce.

Reply:
Apologies for the inconvenience @mborsz. I started a revert, maybe you could take a look: https://github.com/kubernetes/test-infra/pull/16733.

Looking again at the default deployment path, is it safe to assume --deployment=bash only supports the gce provider currently? (i.e., this behavior is correct barring that i've put "gce" and not "bash" in the switch)

Reply:
My understanding is that right now gce is an only supported provider in deplyment=bash mode, but I'm not sure if we want to use this assumption here (what if some day someone adds support for some other provider)?

Personally, I would go with
```
if deployment == "aks" {
   use logic specific for aks
} else if deployment == "bash" and provider == "gce" {
   use logic for gce
} else {
  fatal error
}
```
What do you think about this solution?

Reply:
Makes sense to me. I’m happy to make this change on top of the revert, assuming that’s the sequence you’d prefer.

I wonder if there is some CI automation we can improve to avoid this situation in the future?